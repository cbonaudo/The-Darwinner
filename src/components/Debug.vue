<template>
  <div class="debug">
    <button @click="reset">Reset</button>
    <p>Clicks : {{ clicks }}</p>
    <p>Key Strokes : {{ keyStrokes }}</p>
    <p>Timer : {{ timer }}</p>
  </div>
</template>

<script>
export default {
  name: 'Goal',
  data() {
    return { eraNumber: 1, timer: 0 }
  },
  props: {
    clicks: { default: 0 },
    keyStrokes: { default: 0 }
  },
  created() {
    setInterval(this.tick, 1000)
  },
  methods: {
    tick() {
      this.timer++
    },
    reset() {
      this.$emit('resetAll')
      this.timer = 0
    }
  }
}
</script>

<style>
.debug {
  border: 1px black solid;
  margin: auto 75px;
  padding: 10px;
}
</style>
