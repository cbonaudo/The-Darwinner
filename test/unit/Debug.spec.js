import Debug from '@/components/Debug'
import { shallowMount } from '@vue/test-utils'

describe('Debug.vue', () => {
  const wrapper = shallowMount(Debug)

  it('should render correct contents', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have correct starting data', () => {
    expect(Debug.data()).toMatchSnapshot()
  })

  describe('reset', () => {
    it('should reset the timer', () => {
      wrapper.setData({ timer: 10 })
      expect(wrapper.vm.timer).toEqual(10)
      wrapper.vm.reset()
      expect(wrapper.vm.timer).toEqual(0)
    })
  })

  describe('tick', () => {
    it('should increase the timer', () => {
      wrapper.setData({ timer: 0 })
      expect(wrapper.vm.timer).toEqual(0)
      wrapper.vm.tick()
      expect(wrapper.vm.timer).toEqual(1)
    })
  })
})
