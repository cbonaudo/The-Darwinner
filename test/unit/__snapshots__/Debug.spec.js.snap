// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debug.vue should have correct starting data 1`] = `
Object {
  "eraNumber": 1,
  "timer": 0,
}
`;

exports[`Debug.vue should render correct contents 1`] = `
<div
  class="debug"
>
  <button>
    Reset
  </button>
   
  <p>
    Clicks : 0
  </p>
   
  <p>
    Key Strokes : 0
  </p>
   
  <p>
    Timer : 0
  </p>
</div>
`;
